homeassistant/components/file/__init__.py

```diff
@@ -76,11 +78,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             translation_placeholders={"filename": filepath},
         )
 
-    if entry.data[CONF_PLATFORM] in PLATFORMS:
-        await hass.config_entries.async_forward_entry_setups(
-            entry, [Platform(entry.data[CONF_PLATFORM])]
-        )
-    else:
+    await hass.config_entries.async_forward_entry_setups(
+        entry, [Platform(entry.data[CONF_PLATFORM])]
+    )
+    if entry.data[CONF_PLATFORM] == Platform.NOTIFY and CONF_NAME in entry.data:
         # The notify platform is not yet set up as entry, so
         # forward setup config through discovery to ensure setup notify service.
```

Comment:
Update the comment to explain that we want to keep the legacy notify service around during the deprecation period, even though we now support notify config entry.